modules/build/src/main/scala/scala/build/ConsoleBloopBuildClient.scala

```diff
@@ -204,4 +208,29 @@ object ConsoleBloopBuildClient {
     }
   }
 
+  def printOtherDiagnostic(
+    out: PrintStream,
+    message: String,
+    severity: Severity,
+    positions: Seq[Position]
+  ): Unit = {
+    val isWarningOrError = true
+    if (isWarningOrError) {
+      val msgIt  = message.linesIterator
+      val prefix = getPrefix(severity)
+      out.println(prefix + (if (msgIt.hasNext) " " + msgIt.next() else ""))
+      msgIt.foreach(line => out.println(prefix + line))
+
+      positions.foreach { pos =>
+        pos match {
+          case Position.Bloop(bloopJavaPath) =>
+            val bloopOutputPrefix = s"[current bloop jvm] "
+            out.println(prefix + bloopOutputPrefix + bloopJavaPath)
+            out.println(prefix + " " * bloopOutputPrefix.length + "^" * bloopJavaPath.length())
+          case pos => out.println(prefix + pos.render())
+        }
+      }
+    }
```

Comment:
NIT (more idiomatic)
```suggestion
      positions.foreach {
        case Position.Bloop(bloopJavaPath) =>
          val bloopOutputPrefix = s"[current bloop jvm] "
          out.println(prefix + bloopOutputPrefix + bloopJavaPath)
          out.println(prefix + " " * bloopOutputPrefix.length + "^" * bloopJavaPath.length())
        case pos => out.println(prefix + pos.render())
      }
    }
```